pyCGM_Single/tests/test_Pipelines.py

```diff
@@ -0,0 +1,517 @@
+"""
+#Test positive and negative floats and ints
+listTests = [
+        [-1003.58361816, -1003.50396729, -1003.42358398, -1003.3425293,
+        -1003.26068115, -1003.17810059, -1003.09484863, -1003.01080322,
+        -1002.92608643, -1002.84063721, -1002.75445557, -1002.6675415,
+        -1002.57995605, -1002.49157715, -1002.40252686, -1002.31274414],
+        [70, 76, 87, 52, 28, 36, 65, 95, 69, 62, 60, 17, 12, 97, 11, 48],
+        [-70, 76, 87, -52, 28, 36, 65, 95, -69, 62, 60, 17, 12, -97, 11, 48]
+]
+
+#Same input as listTests, but as numpy arrays
+numpyTests = [np.array(arr) for arr in listTests]
+
+#Expected results are the same for both lists and numpy arrays
+expectedResults = [
+    np.array([-1003.58359202, -1003.50390817, -1003.42363665, -1003.34252947,
+              -1003.26066539, -1003.17811173, -1003.09483484, -1003.01081474,
+              -1002.92608178, -1002.8406421 , -1002.7544567 , -1002.66753023,
+              -1002.57993488, -1002.49164743, -1002.40251014, -1002.31267097]),
+
+    np.array([69.98542399, 84.63525626, 78.90545928, 52.35480234, 
+              30.49005341, 37.66166803, 65.18623797, 84.35193788, 
+              82.07291528, 64.17505539, 41.24155139, 28.10878834,
+              33.98739195, 47.41539095, 52.2509855,  48.00079742]),
+
+    np.array([-69.96417558, 62.8459697, 68.01436426, -1.36719047,
+              -12.10645143, 48.64339139,  81.14625417,  41.74781614,
+              1.18057508, 21.16904888, 57.9718148, 39.85271513, 
+              -21.05573085, -53.81198321, -23.50929002, 47.94081346])
+]
+@pytest.mark.parametrize("data, cutoff, Fs, expectedResult", [
+    (listTests[0], 20, 120, expectedResults[0]),
+    (listTests[1], 20, 120, expectedResults[1]),
+    (listTests[2], 20, 120, expectedResults[2]),
+    (np.array(listTests[0]), 20, 120, expectedResults[0]),
+    (np.array(listTests[1]), 20, 120, expectedResults[1]),
+    (np.array(listTests[2]), 20, 120, expectedResults[2])
+])
+def test_butterFilterAccuracy(data, cutoff, Fs, expectedResult):
+    #Call butterFilter() with the values from listTests and numpyTests
+    result = Pipelines.butterFilter(data, cutoff, Fs)
+    np.testing.assert_almost_equal(result, expectedResult, rounding_precision)
+
+@pytest.mark.parametrize("data, cutoff, Fs", [
+    #Test that if len(data) < 15, an exception is raised
+    ([], 20, 120),
+    (range(3), 20, 120),
+    (range(15), 20, 120),
+    (np.arange(0, 0.4, 0.1), 20, 120),
+    #Test invalid values for cutoff frequency will raise exceptions
+    (range(20), -10, 120),
+    (range(20), 0, 120),
+    (range(20), 40, 120),
+    #Test invalid values for Fs, the sampling frequency will raise exceptions
+    (range(20), 20, -10),
+    (range(20), 20, 0),
+    (range(20), 20, 20),
+    (range(20), 20, 60)
+])
+def test_butterFilterExceptions(data, cutoff, Fs):
+    with pytest.raises(Exception):
+        Pipelines.butterFilter(data, cutoff, Fs)
+
+"""
```

Comment:
What is this string for? It seems like a docstring, but is below one function and above another.  It may be clearer to use it as a docstring.

Reply:
Changed the string to appear inside of the testing functions so it is clearer as a docstring.